FirebaseDatabase/Tests/Integration/FIRDatabaseQueryTests.m

```diff
@@ -3056,4 +3060,192 @@ - (void)testAddingListensForTheSamePathDoesNotCheckFail {
   WAIT_FOR(done);
 }
 
+- (void)testEmptyQueryGet {
+  FIRDatabaseReference* ref = [FTestHelpers getRandomNode];
+
+  __block BOOL done = NO;
+
+  [ref getDataWithCompletionBlock:^(NSError* err, FIRDataSnapshot* snapshot) {
+    XCTAssertNil(snapshot.value);
+    done = YES;
+  }];
+
+  WAIT_FOR(done);
+}
+
+- (void)testOfflineQueryGet {
+  FIRDatabase* db = [FTestHelpers defaultDatabase];
+  FIRDatabaseReference* ref = [db reference];
+
+  __block BOOL done = NO;
+
+  [db goOffline];
+
+  [ref getDataWithCompletionBlock:^(NSError* err, FIRDataSnapshot* snapshot) {
+    XCTAssertNotNil(err);
+    XCTAssertEqualObjects([err localizedFailureReason], kPersistentConnOffline);
+    done = YES;
+  }];
+
+  WAIT_FOR(done);
+}
+
+- (void)testGetQueryBasic {
+  FIRDatabase* db = [FTestHelpers defaultDatabase];
+  FIRDatabaseReference* ref = [db reference];
+
+  __block BOOL done = NO;
+
+  [ref setValue:@42
+      withCompletionBlock:^(NSError* error, FIRDatabaseReference* ref) {
+        XCTAssertNil(error);
+        done = YES;
+      }];
+
+  WAIT_FOR(done);
+
+  [ref getDataWithCompletionBlock:^(NSError* err, FIRDataSnapshot* snapshot) {
+    XCTAssertNil(err);
+    XCTAssertEqualObjects(snapshot.value, @42);
+    done = YES;
+  }];
+
+  WAIT_FOR(done);
+}
+
+- (void)testQueryGetCached {
+  FIRDatabase* db = [FTestHelpers defaultDatabase];
+  FIRDatabaseReference* ref = [db reference];
+
+  __block BOOL done = NO;
+
+  [ref observeEventType:FIRDataEventTypeValue
+              withBlock:^(FIRDataSnapshot* snapshot) {
+                id value = [snapshot value];
+                if (value != nil && [value isEqualToNumber:@42]) {
+                  done = YES;
+                }
+              }];
+
+  WAIT_FOR(done);
```

Comment:
How does this work here? We are only setting the value after this `WAIT_FOR` call completes.

Reply:
Thanks for catching this.